docusaurus/docs/tutorials/ui-extensions/register-an-extension.md

```diff
@@ -0,0 +1,235 @@
+
+### Register a component
+
+```tsx
+import { PluginExtensionPoints } from '@grafana/data';
+
+export const plugin = new AppPlugin().addComponent({
+  title: 'User profile tab',
+  description: '...',
+  targets: [PluginExtensionPoints.UserProfileTab],
+  component: () => <div>This is a new tab on the user profile page.</div>,
+});
+```
+
+### Accessing plugin meta in a component
+
+You can use the `usePluginContext()` hook to access any plugin specific meta information inside your component. The hook [returns a `PluginMeta` object](https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/types/plugin.ts#L62). (This can be useful because the component that you register from your plugin won't be rendered under the React tree of your plugin, but somewhere else in the UI.)
+
+```tsx
+import { usePluginContext, PluginExtensionPoints } from '@grafana/data';
+
+export const plugin = new AppPlugin().addComponent({
+  title: 'User profile tab',
+  description: '...',
+  targets: [PluginExtensionPoints.UserProfileTab],
+  component: () => {
+    const { meta } = usePluginContext();
+
+    // The `jsonData` property is an object that your plugin can manage
+    // using the Grafana Rest APIs
+    return <div>Plugin specific setting: {meta.jsonData.foo}</div>;
+  },
+});
+```
+
+### Access plugin state in a component
+
+```tsx
+import { PluginExtensionPoints } from '@grafana/data';
+import { MyCustomDataProvider } from './MyCustomDataProvider';
+
+export const plugin = new AppPlugin().addComponent({
+  title: 'User profile tab',
+  description: '...',
+  targets: [PluginExtensionPoints.UserProfileTab],
+  component: () => (
+    <MyCustomDataProvider>
+      <div>Plugin specific setting: {meta.jsonData.foo}</div>
+    </MyCustomDataProvider>
+  ),
+});
+```
+
+### Hide a component in certain conditions
+
+This is nothing new for developers working with React, just return `null` from your component in order to not render anything (to hide it).
+
+```tsx
+import { usePluginContext, PluginExtensionPoints } from '@grafana/data';
+
+export const plugin = new AppPlugin().addComponent({
+  title: 'User profile tab',
+  description: '...',
+  targets: [PluginExtensionPoints.UserProfileTab],
+  component: () => {
+    const { meta } = usePluginContext();
+
+    // For the sake of the example this condition is relying on a `jsonData` property
+    // that is managed by your plugin
+    if (!meta.jsonData.isExtensionEnabled) {
+      return null;
+    }
+
+    return <div>Plugin specific setting: {meta.jsonData.foo}</div>;
+  },
+});
+```
+
+## Troubleshooting
+
+### My link is not appearing
+
+1. **Check the console logs** - there is a good chance that your link is not appearing due to some validation errors. In this case you should see some relevant logs in your browsers console.
```

Comment:
Maybe give an example of a log line?

Reply:
I think that's a good idea, yeah. @mckn which log you think we should add as an example?

Reply:
Probably the one with an invalid extension point id.

Reply:
We tried adding a log example, but it's hard to get this to look good since we're using a numbered list. Let's skip this for now.